Controllers/AdminController.cs

```diff
@@ -51,24 +63,48 @@ public IActionResult Index(long id)
         public IActionResult CreateBook()
         {
             //Form uses the collection of Writers to populate the drop-down list
-            return View(new CreateBookVM { Writers = _db.Writers });
+            return View(new CreateBookVM { Writers = _db.Writers.OrderBy(b => b.Name) });
         }
         [HttpPost]
         public IActionResult CreateBook(CreateBookVM bookVM, long id)
         {
             Writer writer;
+            if (bookVM.AuthorId > 0)
+            {
+                writer = _db.Writers.Single(b => b.Id == bookVM.AuthorId);
+            }
+            else
+            {
+                writer = new Writer { Name = bookVM.Name };
+                _db.Add(writer);
+                _db.SaveChanges();
```

Comment:
Calling `SaveChanges()` here is fine, but you can wait to do this only ONCE after you have added both the writer and the book